intermediate_source/custom_function_conv_bn_tutorial.py

```diff
@@ -0,0 +1,344 @@
+
+class Net(nn.Module):
+    def __init__(self, fused=True):
+        super(Net, self).__init__()
+        self.fused = fused
+        if fused:
+            self.convbn1 = FusedConvBN(1, 32, 3)
+            self.convbn2 = FusedConvBN(32, 64, 3)
+            self.convbn3 = FusedConvBN(64, 64, 3)
+        else:
+            self.conv1 = nn.Conv2d(1, 32, 3, 1, bias=False)
+            self.bn1 = nn.BatchNorm2d(32)
+            self.conv2 = nn.Conv2d(32, 64, 3, 1, bias=False)
+            self.bn2 = nn.BatchNorm2d(64)
+            self.conv3 = nn.Conv2d(64, 64, 3, 1)
+            self.bn3 = nn.BatchNorm2d(64)
+        self.fc1 = nn.Linear(9216, 128)
+        self.dropout = nn.Dropout(0.5)
+        self.fc2 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        if self.fused:
+            x = self.convbn1(x)
+        else:
+            x = self.conv1(x)
+            x = self.bn1(x)
+        x = F.relu(x)
+        if self.fused:
+            x = self.convbn2(x)
+        else:
+            x = self.conv2(x)
+            x = self.bn2(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = F.relu(x)
+        x = x.flatten(1)
+        x = self.fc1(x)
+        x = self.dropout(x)
+        x = F.relu(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
+        return output
+
+def train(model, device, train_loader, optimizer, epoch):
+    model.train()
+    for batch_idx, (data, target) in enumerate(train_loader):
+        data, target = data.to(device), target.to(device)
+        optimizer.zero_grad()
+        output = model(data)
+        loss = F.nll_loss(output, target)
+        loss.backward()
+        optimizer.step()
+        if batch_idx % 2 == 0:
+            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
+                epoch, batch_idx * len(data), len(train_loader.dataset),
+                100. * batch_idx / len(train_loader), loss.item()))
+
+def test(model, device, test_loader):
+    model.eval()
+    test_loss = 0
+    correct = 0
+    # Use inference mode instead of no_grad, for free improved test-time performance
+    with torch.inference_mode():
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            test_loss += F.nll_loss(output, target, reduction='sum').item()  # sum up batch loss
+            pred = output.argmax(dim=1, keepdim=True)  # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+
+    print('\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n'.format(
+        test_loss, correct, len(test_loader.dataset),
+        100. * correct / len(test_loader.dataset)))
+
+use_cuda = torch.cuda.is_available()
+
+torch.manual_seed(123456)
+
+device = torch.device("cuda" if use_cuda else "cpu")
+
+train_kwargs = {'batch_size': 4096}
+test_kwargs = {'batch_size': 1024}
```

Comment:
Why is the test batch size bigger than the train one? You should be able to test more since you don't save any buffer